src/reason_pprint_ast.ml

```diff
@@ -5065,6 +5065,36 @@ class printer  ()= object(self:'self)
     | Pstr_attribute (s, _) -> (not (s.txt = "ocaml.text") && not (s.txt = "ocaml.doc"))
     | _ -> true
 
+  (*
+    external render : reactElement => element => unit =   (* frstHalf *)
+      "render" [@@bs.val] [@@bs.module "react-dom"];      (* sndHalf *)
+
+    To improve the formatting with breaking & indentation:
+      * consider the part before the '=' as a label
+      * combine that label with '=' in a list
+      * consider the part after the '=' as a list
+      * combine both parts as a label
+  *)
+  method primitive_declaration vd =
+    let attrs = List.map (fun x -> self#item_attribute x) vd.pval_attributes in
+    let lblBefore =
+      label ~space:true
+        (makeList ~postSpace:true
+          [atom "external"; protectIdentifier vd.pval_name.txt; atom ":"])
+        (self#core_type vd.pval_type)
+    in
+    let frstHalf = makeList ~postSpace:true [lblBefore; atom "="] in
+    let string_literals = makeSpacedBreakableInlineList (List.map self#constant_string vd.pval_prim) in
+    (* when there aren't any attrs, add a final semi here *)
+    let prim = if (List.length attrs) == 0
+      then appendSep false ";" string_literals
```

Comment:
Took me a while to understand why you add a ";" here.  I think it is probably due to some tweaks we put into comment interleaving logic?